packages/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx

```diff
@@ -0,0 +1,615 @@
+      }
+      const ytooltipProps = {
+        tooltipCls: this._classNames.tooltip!,
+        id: this._tooltipId,
+        yAxis: yAxisElement,
+      };
+      yAxisElement && this._tooltipOfYAxislabels(ytooltipProps);
+    }
+    return bars;
+  }
+
+  private _closeCallout = () => {
+    this.setState({
+      isCalloutVisible: false,
+    });
+  };
+
+  private _onLegendClick(customMessage: string): void {
+    if (this.state.isLegendSelected) {
+      if (this.state.selectedLegendTitle === customMessage) {
+        this.setState({
+          isLegendSelected: false,
+          selectedLegendTitle: customMessage,
+        });
+      } else {
+        this.setState({
+          selectedLegendTitle: customMessage,
+        });
+      }
+    } else {
+      this.setState({
+        isLegendSelected: true,
+        selectedLegendTitle: customMessage,
+      });
+    }
+  }
+
+  private _onLegendHover(customMessage: string): void {
+    if (this.state.isLegendSelected === false) {
+      this.setState({
+        isLegendHovered: true,
+        selectedLegendTitle: customMessage,
+      });
+    }
+  }
+
+  private _onLegendLeave(isLegendFocused?: boolean): void {
+    if (!!isLegendFocused || this.state.isLegendSelected === false) {
+      this.setState({
+        isLegendHovered: false,
+        selectedLegendTitle: '',
+        isLegendSelected: isLegendFocused ? false : this.state.isLegendSelected,
+      });
+    }
+  }
+
+  private _getLegendData = (data: IHorizontalBarChartWithAxisDataPoint[], palette: IPalette): JSX.Element => {
+    const { useSingleColor } = this.props;
+    const actions: ILegend[] = [];
+    data.forEach((point: IHorizontalBarChartWithAxisDataPoint, _index: number) => {
+      const color: string = !useSingleColor ? point.color! : this._createColors()(1);
+      // mapping data to the format Legends component needs
+      const legend: ILegend = {
+        title: point.legend!,
+        color: color,
+        action: () => {
+          this._onLegendClick(point.legend!);
+        },
+        hoverAction: () => {
+          this._onLegendHover(point.legend!);
+        },
+        onMouseOutAction: (isLegendSelected?: boolean) => {
+          this._onLegendLeave(isLegendSelected);
+        },
+      };
+      actions.push(legend);
+    });
+    const legends = (
+      <Legends
+        legends={actions}
+        enabledWrapLines={this.props.enabledLegendsWrapLines}
+        overflowProps={this.props.legendsOverflowProps}
+        focusZonePropsInHoverCard={this.props.focusZonePropsForLegendsInHoverCard}
+        overflowText={this.props.legendsOverflowText}
+        {...this.props.legendProps}
+      />
+    );
+    return legends;
+  };
+
+  private _getAxisData = (yAxisData: IAxisData) => {
+    if (yAxisData && yAxisData.yAxisDomainValues.length) {
+      const { yAxisDomainValues: domainValue } = yAxisData;
+      this._xMax = Math.max(domainValue[domainValue.length - 1], this.props.xMaxValue || 0);
```

Comment:
There are places where x and y are swapped. Lets add comment wherever applicable to explain it.

Reply:
Done